.github/actions/generate-coverage/tests/conftest.py

```
@@ -0,0 +1,35 @@
+from pathlib import Path
+
+import pytest
+
+
+def _find_root(start: Path) -> Path:
+    """Return the nearest ancestor containing ``pyproject.toml`` or ``.git``."""
+
+    for parent in (start, *start.parents):
+        if (parent / "pyproject.toml").exists() or (parent / ".git").exists():
+            return parent
+    raise FileNotFoundError(f"Project root not found from {start}")
```

Ruff (0.11.9)

Avoid specifying long messages outside the exception class

(TRY003)